packages/gestalt/src/Dropdown.js

```diff
@@ -220,8 +222,18 @@ export default function Dropdown({
       role="menu"
       shouldFocus
       size="xl"
+      __dangerouslyIgnoreScrollBoundaryContainerSize={
+        __dangerouslyIgnoreScrollBoundaryContainerSize
+      }
     >
-      <Box alignItems="center" direction="column" display="flex" flex="grow" margin={2}>
+      <Box
+        alignItems="center"
+        direction="column"
+        display="flex"
+        flex="grow"
+        margin={2}
+        maxHeight={__dangerouslyIgnoreScrollBoundaryContainerSize ? '30vh' : 'auto'}
```

Comment:
let's leave        ` maxHeight={__dangerouslyIgnoreScrollBoundaryContainerSize ? '30vh' : undefined} `

I'm not sure if auto equals undefined here, therefore, let's leave the equivalent.

![Screenshot 2023-03-06 at 6 14 54 PM](https://user-images.githubusercontent.com/10593890/223278135-e673fe03-8c0e-45fe-805e-40434d183134.png)

Reply:
Or maybe somehting like         {...(lineClamp ? { style: { WebkitLineClamp: lineClamp } } : {})}

Reply:
Also, add comment on why we are doing this here. It's undocumented but we should leave a description for internal knowldge

// __dangerouslySetMaxWidth can be set when `ScrollBoundaryContainer` is set to overflow="visible" but therefore limits the height of the Popover-based component.  Some cases require

Reply:
The failing snapshots below are an example of Alberto's concerns with using `'auto'` instead of `undefined`

Reply:
If we follow Ryan's comment, then we should document the prop https://github.com/pinterest/gestalt/pull/2675#discussion_r1127246052

Which honestly makes sense not keeping undocumented.

Let's not add comment but document the prop "prop"erly  xd